cosmwasm/contracts/token-bridge/src/contract.rs

```diff
@@ -1208,16 +1214,22 @@ fn handle_initiate_transfer_token(
 
             // chop off dust
             amount = Uint128::new(
-                amount
+                match amount
                     .u128()
                     .checked_sub(amount.u128().checked_rem(multiplier).unwrap())
-                    .unwrap(),
+                {
+                    None => return Result::Err(cosmwasm_std::StdError::generic_err("Insufficient Funds")),
+                    Some(value) => value
+                }
             );
 
             fee = Uint128::new(
-                fee.u128()
+                match fee.u128()
                     .checked_sub(fee.u128().checked_rem(multiplier).unwrap())
```

Comment:
I think we probably don't need checked math for div and rem operations because those can only fail when the right hand side is 0, which is never the case in the code so far.  We can still leave them in case we change the multiplier in the future.  If we want to make this more ergonomic we can do something like:

```rust
    fee = fee
        .u128()
        .checked_rem(multiplier)
        .and_then(|rem| fee.u128().checked_sub(rem))
        .map(Uint128::new)
        .ok_or_else(|| StdError::generic_err("Invalid fee"))?;
```